lib/src/executor_layer/error.rs

```diff
@@ -0,0 +1,25 @@
+use super::super::storage_layer::error::StorageError;
+
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum ExecutorError {
+    #[error("Message type mismatch")]
+    MessageTypeMismatch,
+
+    #[error("Table already exists")]
+    TableAlreadyExists,
+
+    #[error("Table not found")]
+    TableNotFound,
+
+    #[error("Underlying error: {0}")]
```

Comment:
You can use thiserror's `#[from]` here

Reply:
It would conflict with From impl for marble and bincode errors